pennylane/hf/dipole.py

```diff
@@ -0,0 +1,310 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed for computing the dipole moment.
+"""
+import autograd.numpy as anp
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.hf.basis_data import atomic_numbers
+from pennylane.hf.hamiltonian import _generate_qubit_operator, _return_pauli, simplify
+from pennylane.hf.matrices import moment_matrix
+
+
+def dipole_integrals(mol, core=None, active=None):
+    r"""Return a function that computes the dipole moment integrals over the molecular orbitals.
+
+    These integrals are required to construct the dipole operator in the second-quantized form
+
+    .. math::
+
+        D = \sum_{pq} d_{pq} c_p^{\dagger} c_q + D_n,
+
+    where the coefficients :math:`d_{pq}` are given by the integral over molecular orbitals
+    :math:`\phi`
+
+    .. math::
+
+        d_{pq} = \int \phi_p^*(r) r \phi_q(r) dr,
+
+    :math:`c^{\dagger}` and :math:`c` are the creation and annihilation operators, respectively, and
+    :math:`D_n` is the contribution of the nuclei to the dipole operator.
+
+    The molecular orbitals are represented as a linear combination of atomic orbitals as
+
+    .. math::
+
+        \phi_i(r) = \sum_{\nu}c_{\nu}^i \chi_{\nu}(r).
+
+    Using this equation the dipole moment integral :math:`d_{pq}` can be written as
+
+    .. math::
+
+        d_{pq} = \sum_{\mu \nu} C_{p \mu} d_{\mu \nu} C_{\nu q},
+
+    where :math:`d_{\mu \nu}` is the dipole moment integral over the atomic orbitals and :math:`C`
+    is the molecular orbital expansion coefficient matrix.
+
+    Args:
+        mol (Molecule): the molecule object
+        core (list[int]): indices of the core orbitals
+        active (list[int]): indices of the active orbitals
+
+    Returns:
+        function: function that computes the dipole moment integrals in the molecular orbital basis
+
+    **Example**
+
+    >>> symbols  = ['H', 'H']
+    >>> geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.0]], requires_grad = False)
+    >>> alpha = np.array([[3.42525091, 0.62391373, 0.1688554],
+    >>>                   [3.42525091, 0.62391373, 0.1688554]], requires_grad=True)
+    >>> mol = qml.hf.Molecule(symbols, geometry, alpha=alpha)
+    >>> args = [alpha]
+    >>> constants, integrals = dipole_integrals(mol)(*args)
+    >>> print(integrals)
+    (array([[0., 0.],
+            [0., 0.]]),
+     array([[0., 0.],
+            [0., 0.]]),
+     array([[ 0.5      , -0.8270995],
+            [-0.8270995,  0.5      ]]))
+    """
+
+    def _dipole_integrals(*args):
+        r"""Compute the dipole moment integrals in the molecular orbital basis.
+
+        Args:
+            args (array[array[float]]): initial values of the differentiable parameters
+
+        Returns:
+            tuple[array[float]]: tuple containing the contributions due to the core orbitals and the
+            nuclei and the dipole moment integrals
```

Comment:
Double "and" in the same sentence, please revise

Reply:
Corrected. The contribution of the nuclei is now computed in  fermionic_dipole, which makes more sense.